cranelift/codegen/src/lib.rs

```diff
@@ -97,9 +97,8 @@ pub mod incremental_cache;
 #[macro_export]
 macro_rules! trace {
     ($($tt:tt)*) => {
-        if cfg!(feature = "trace-log") {
-            ::log::trace!($($tt)*);
-        }
+        #[cfg(any(feature = "trace-log", debug_assertions))]
```

Comment:
Here you'll want to use `if cfg!(...) { ... }` instead of `#[cfg(...)]` because that's what's causing the failure in CI. With `#[cfg]` then in `--release` build mode these statements are compiled out meaning that if any variables or such are only used for the log statements then they're all of a sudden unused variables and get code warnings. By using `if cfg!(...)` it keeps the same shape of the code in release/debug mode and the consequence is that the optimizer has to do a little more work to determine that it's the same as the `#[cfg]` mode

Reply:
Thank you!